breakevenApi/breakevenApi/Controllers/DoencaController.cs

```diff
@@ -0,0 +1,81 @@
+﻿using breakevenApi.Domain.Entities.Doenca;
+using Microsoft.AspNetCore.Mvc;
+
+namespace breakevenApi.Controllers
+{
+    [Route("doenca")]
+    public class DoencaController : Controller
+    {
+        private readonly IDoencaRepository _doencaRepository;
+
+        public DoencaController(IDoencaRepository doencaRepository)
+        {
+            _doencaRepository = doencaRepository;
+        }
+
+
+        [HttpGet]
+        [Route("get/")]
+        public IActionResult GetDoencaById([FromQuery] long idDoenca)
+        {
+            try
+            {
+                var doenca = _doencaRepository.GetById(idDoenca);
+                if (doenca == null) return NotFound("Doenca nao encontrada");
+                return Ok(doenca);
+            
+            }
+            catch(Exception e)
+            {
+
+               return BadRequest(e.Message);
+            }        
+        }
+
+        [HttpGet]
+        [Route("get-all")]
+        public IActionResult GetAllDoencas()
+        {
+            try
+            {
+                return Ok(_doencaRepository.GetAll());
+            }
+            catch(Exception e)
+            {
+                return BadRequest(e.Message);
+            }
+        }
```

Comment:
Não é necessário alterar, mas um ponto sobre essa abordagem que pode ser um pouco ruim pensando na observabilidade da api, está sendo tratado uma Exception genérica, ou seja, qualquer erro que a sua api der vai ser convertida a um erro 400 (que deve ser utilizado quando o request enviado está incorreto), o ideal nesses casos é tratarmos exceptions especializadas ao invés de genéricas 

Ref

https://learn.microsoft.com/pt-br/dotnet/standard/exceptions/best-practices-for-exceptions